starlark/src/values/error.rs

```diff
@@ -84,6 +84,16 @@ pub enum ValueError {
     TypeNotSupported(String),
 }
 
+/// Returned when Starlark evaluation results in too deep recursion.
```

Comment:
Why do we need a separate type here for the error?

Reply:
To make it clear that functions are not allowed to return any other error. Types must be comparable for equality.